viewing_party/party.py

```diff
@@ -1,23 +1,145 @@
+    # loop through movies in watched and take out the most popular genre
+    # add to dict and use a counter 
+    genre_dict = {}
+
+    if len(user_data["watched"]) == 0:
+        return None
+    else:
+        for movie in user_data["watched"]:
+            genre = movie["genre"]
+            if genre not in genre_dict:
+                genre_dict[genre] = 1
+            else:
+                genre_dict[genre] += 1
+
+    #loop through genre_dict and have 2 variables (count,genre)
+    most_genre_count = 0
+    most_genre = ""
+    for genre, genre_times in genre_dict.items():
+        if genre_times > most_genre_count:
+            most_genre = genre
+            most_genre_count = genre_times
+
+
+    return most_genre
+        
 # -----------------------------------------
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def  get_unique_watched(user_data):
+    unique_list = []
+    friends_watched = get_friends_watched(user_data)
+    for movie in user_data["watched"]:
+        if movie not in friends_watched and movie not in unique_list:
+            unique_list.append(movie)
+    # print(unique_list)
+    return unique_list
 
-        
+def get_friends_watched(user_data):
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movies in friend.values():
+            for movie in movies:
+                friends_watched.append(movie)
+    return friends_watched
+
+def get_friends_unique_watched(user_data):
+    friends_watched = get_friends_watched(user_data)
+    friends_unique = []
+    for movie in friends_watched:
+        if movie not in user_data["watched"] and movie not in friends_unique:
+            friends_unique.append(movie) 
+    return friends_unique       
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
-
+def get_available_recs(user_data):
+    friends_watched = get_friends_watched(user_data)
+    rec = []
+    for movie in friends_watched:
+        if movie not in user_data["watched"] and movie in friends_watched and movie["host"] in user_data["subscriptions"]:
+            rec.append(movie)
+    return rec
+        
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    get_genres = get_most_watched_genre(user_data)
+    friends_watched = get_friends_unique_watched(user_data)
+    genre = []
+    
+    for movie in friends_watched:
+        if movie["genre"] == get_genres:
+            genre.append(movie)
+    return genre
+
+
+def get_rec_from_favorites(user_data):
+    rec_from_favs = []
+
+    user_unique = get_unique_watched(user_data)
+
+    for movie in user_unique:
+        if movie in user_data["favorites"]:
+            rec_from_favs.append(movie)
+            
+    print(rec_from_favs)

```

Comment:
We can remove this print statement since it's not part of the function requirements in the readme.